rust/kernel/sync/arc.rs

```diff
@@ -30,7 +34,7 @@ use core::{
 /// # Invariants
 ///
 /// The value stored in [`RefCounted::get_count`] corresponds to the number of instances of [`Ref`]
-/// that point to that instance of `T`.
+/// that point to that instance of `T`. So it is always non-zero when a [`Ref`] exists.
```

Comment:
Since this is the type invariant for `Ref`, the "when a `Ref` does not exist" case cannot happen, i.e. it is always non-zero. So perhaps I would clarify that the "when" is the reason, rather than a condition:

```suggestion
/// that point to that instance of `T`. So it is always non-zero (because [`Ref`] exists).
```

Reply:
Done.